jnigen/lib/src/bindings/dart_generator.dart

```diff
@@ -0,0 +1,1111 @@
+const _classRef = '_classRef';
+const _env = 'jniEnv';
+const _accessors = 'jniAccessors';
+const _lookup = 'jniLookup';
+const _selfPointer = 'reference';
+
+// Docs
+const _deleteInstruction =
+    '  /// The returned object must be deleted after use, '
+    'by calling the `delete` method.';
+
+extension _StringX on String {
+  String capitalize() {
+    return '${this[0].toUpperCase()}${substring(1)}';
+  }
+}
+
+/// Encloses [inside] in the middle of [open] and [close]
+/// if [inside] is not empty.
+String _encloseIfNotEmpty(String open, String inside, String close) {
+  if (inside == '') return '';
+  return '$open$inside$close';
+}
+
+String _newLine({int depth = 0}) {
+  return '\n${'  ' * depth}';
+}
+
+// **Naming Convention**
+//
+// Let's take the following code as an example:
+//
+// ```dart
+// Method definition
+// void f<T extends num, U>(JType<T> $T, JType<U> $U, T t, U u) {
+//   // ...
+// }
+// f<int, String>($T, $U, t, u); // Calling the Method
+// ```
+//
+// Here `f` will be replaced according to the place of usage.
+//
+// * `fArgsDef` refers to `T t, U u` – the arguments in the method
+//   definition.
+// * `fArgsCall` refer to `t, u` – the arguments passed to call the method.
+// * `fTypeParamsDef` refers to `<T extends num, U>` – the type parameters
+//   of the method at the point of definition.
+// * `fTypeParamsCall` refers to `<int, String>` – the type parameters when
+//   calling, or whenever we don't want to have the `extends` keyword.
+// * `fTypeClassesDef` refers to `JType<T> $T, JType<U> $U`.
+// * `fTypeClassesCall` refers to `$T, $U` when calling the method.
+class DartGenerator extends Visitor<Classes, Future<void>> {
+  DartGenerator(this.config);
+
+  final Config config;
+
+  static const cInitImport = 'import "dart:ffi" as ffi;\n'
+      'import "package:jni/internal_helpers_for_jnigen.dart";\n';
+
+  /// Initialization code for C based bindings.
+  ///
+  /// Should be called once in a package. In package-structured bindings
+  /// this is placed in _init.dart in package root.
+  String get cInitCode => '''
+// Auto-generated initialization code.
+
+final $_ffi.Pointer<T> Function<T extends $_ffi.NativeType>(String sym) $_lookup =
+    ProtectedJniExtensions.initGeneratedLibrary("${config.outputConfig.cConfig!.libraryName}");
+
+
+''';
+
+  static const dartOnlyInitImport = 'import "package:jni/jni.dart" as jni;\n';
+
+  static const dartOnlyInitCode = '''
+// Auto-generated initialization code.
+
+final $_env = $_jni.Jni.env;
+final $_accessors = $_jni.Jni.accessors;
+
+
+''';
+
+  static const autoGeneratedNotice = '// Autogenerated by jnigen. '
+      'DO NOT EDIT!\n\n';
+  static const defaultImports = 'import "dart:isolate" show ReceivePort;\n'
+      'import "dart:ffi" as ffi;\n'
+      'import "package:jni/internal_helpers_for_jnigen.dart";\n'
+      'import "package:jni/jni.dart" as jni;\n\n';
+  static const defaultLintSuppressions =
+      '// ignore_for_file: annotate_overrides\n'
```

Comment:
Please use:

```dart
'''
So that we don't have to
write \n's
''';
```

Reply:
Copied from the old version! Will change!